MultiServer.py

```diff
@@ -503,7 +503,7 @@ def set_save(self, savedata: dict):
 
     def get_hint_cost(self, slot):
         if self.hint_cost:
-            return max(0, int(self.hint_cost * 0.01 * len(self.locations[slot])))
+            return max(0, int(self.hint_cost * 0.01 * len([loc for loc in self.locations[slot].values() if loc[2] & 8 == 0])))
```

Comment:
this makes this normally very fast check (as len is cached) into a much slower version and it gets called a fair bit. A more efficient implementation might be good.